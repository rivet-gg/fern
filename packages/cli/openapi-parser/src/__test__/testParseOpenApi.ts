import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
