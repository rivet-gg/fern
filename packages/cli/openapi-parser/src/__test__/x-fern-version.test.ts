import { testParseOpenAPI } from "./testParseOpenApi";

describe("x-fern-version", () => {
    testParseOpenAPI("x-fern-version", "openapi.yml");
});
