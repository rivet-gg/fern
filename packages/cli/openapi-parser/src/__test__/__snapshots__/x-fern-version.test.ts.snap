// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-fern-version x-fern-version parse open api 1`] = `
{
  "apiVersion": {
    "default": "2.0.0",
    "header": "X-API-Version",
    "values": [
      "1.0.0",
      "2.0.0",
      "Latest",
    ],
  },
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": {
            "properties": {
              "prompt": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "prompt",
                },
              },
            },
            "type": "object",
          },
          "response": {
            "type": "primitive",
            "value": {
              "type": "string",
              "value": "string",
            },
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "ChatCompletionsCreateRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "POST",
      "operationId": null,
      "pagination": null,
      "path": "/completions/chat",
      "pathParameters": [],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "ChatCompletionsCreateRequest",
          "groupName": null,
          "nameOverride": null,
          "schema": "CreateCompletionRequest",
          "type": "reference",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "Success!",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "ChatCompletionsCreateResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": {
            "default": null,
            "format": null,
            "maxLength": null,
            "minLength": null,
            "pattern": null,
            "type": "string",
          },
          "type": "primitive",
        },
        "type": "json",
      },
      "sdkName": {
        "groupName": [
          "chatCompletions",
        ],
        "methodName": "create",
      },
      "server": [],
      "summary": null,
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [],
  "schemas": {
    "Completion": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Completion",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "completionText",
          "key": "text",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "completionText",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "CompletionText",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "CompletionChunk": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "CompletionChunk",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "completionChunkText",
          "key": "text",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "completionChunkText",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "CompletionChunkText",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
    "CreateCompletionRequest": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "CreateCompletionRequest",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "createCompletionRequestStream",
          "key": "stream",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "createCompletionRequestStream",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "CreateCompletionRequestStream",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "type": "boolean",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "createCompletionRequestPrompt",
          "key": "prompt",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "CreateCompletionRequestPrompt",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {
    "access_token": {
      "tokenEnvVar": null,
      "tokenVariableName": null,
      "type": "bearer",
    },
    "api_key": {
      "tokenEnvVar": null,
      "tokenVariableName": null,
      "type": "bearer",
    },
    "custom_header": {
      "headerEnvVar": null,
      "headerName": "X-API-Key",
      "headerVariableName": null,
      "prefix": null,
      "type": "header",
    },
  },
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Test extension \`x-fern-version\`.",
  "variables": {},
  "webhooks": [],
}
`;
